ocaml/xapi/tar_ext.ml

```diff
@@ -0,0 +1,128 @@
+(*
+ * Copyright (c) Cloud Software Group, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Helpers
+
+module D = Debug.Make (struct let name = __MODULE__ end)
+
+open D
+
+let dir_perm = 0o755
+
+let dir_size = 4096L
+
+let inode_size = 4096L
+
+let ( + ) = Int64.add
+
+let ( // ) = Filename.concat
+
+exception Illegal_file_path of string
+
+exception Unsupported_file_type of string
+
+exception File_exceeds_max_size_limit of string
+
+exception File_incomplete of string
+
+exception File_corrupted of string
+
+exception File_parsed_failed of string
+
+let unpack dir acc_size ifd head =
+  let assert_file_path_is_legal file_path =
+    if
+      List.exists
+        (fun s ->
+          String.equal Filename.current_dir_name s
+          || String.equal Filename.parent_dir_name s
+        )
+        (String.split_on_char '/' file_path)
```

Comment:
Could this splitting use the functions in `Filename` also?

Reply:
There is no function for this in `Filename`. Splitting on `/` is safe, though, because there is no escape mechanism in file names and `/` cannot be part of a name.